JavaScriptSPA/auth.js

```diff
@@ -0,0 +1,95 @@
+
+"use strict";
+
+// Browser check variables
+// If you support IE, our recommendation is that you sign-in using Redirect APIs
+// If you as a developer are testing using Edge InPrivate mode, please add "isEdge" to the if check
+const ua = window.navigator.userAgent;
+const msie = ua.indexOf("MSIE ");
+const msie11 = ua.indexOf("Trident/");
+const msedge = ua.indexOf("Edge/");
+const isIE = msie > 0 || msie11 > 0;
+const isEdge = msedge > 0;
+
+let signInType;
+
+// Create the main myMSALObj instance
+// configuration parameters are located at authConfig.js
+const myMSALObj = new Msal.UserAgentApplication(msalConfig);
+
+// Register Callbacks for Redirect flow
+myMSALObj.handleRedirectCallback(authRedirectCallBack);
+
+function authRedirectCallBack(error, response) {
+    if (error) {
+      console.log(error);
+    } else {
+      if (response.tokenType === "id_token" && myMSALObj.getAccount() && !myMSALObj.isCallback(window.location.hash)) {
+        console.log('id_token acquired at: ' + new Date().toString());
+        updateUI();
+        getTokenRedirect(tokenRequest);
+      } else if (response.tokenType === "access_token") {
+        console.log('access_token acquired at: ' + new Date().toString());
+      } else {
+        console.log("token type is:" + response.tokenType);
+      }
+    }
+  }
+
+// Redirect: once login is successful and redirects with tokens, update UI
+if (myMSALObj.getAccount() && !myMSALObj.isCallback(window.location.hash)) {
+    // avoid duplicate code execution on page load in case of iframe and Popup window.
+    updateUI();
+  }
+
+
+function signIn(method) {
+  
+  signInType = isIE ? "Redirect" : method;
```

Comment:
Instead of magic strings, can we use booleans?

Reply:
I've seperated popup and redirect flows into seperate files, so signInType variable is no longer needed